tensorflow/python/ops/math_ops.py

```diff
@@ -99,9 +99,51 @@
 from tensorflow.python.util.tf_export import tf_export
 
 # Aliases for some automatically-generated names.
-linspace = gen_math_ops.lin_space
+#linspace = gen_math_ops.lin_space
 nextafter = gen_math_ops.next_after
 
+def linspace(start_in, stop_in, num, name=None, axis=0):
+  if not isinstance(axis, int):
+    raise TypeError('Axis should be an integer. Received an object {0} of type {1}'.format(axis, type(axis)))
```

Comment:
No need to make your own validation, I think TF's default validation should catch things here.

Reply:
How would it be able to catch it, is there a default validation for all parameters named `axis` somewhere?

Reply:
No, but expand_dims has a parameter named axis and will do this validation for you

Reply:
Aaah, I see. Thanks, will do